src/Microsoft.ML.Data/Transforms/ConcatTransform.cs

```diff
@@ -113,6 +127,218 @@ public sealed class TaggedArguments
             public TaggedColumn[] Column;
         }
 
+        public sealed class ColumnInfo
+        {
+            public readonly string Output;
+
+            private readonly (string name, string alias)[] _inputs;
+            public IReadOnlyList<(string name, string alias)> Inputs => _inputs.AsReadOnly();
+
+            public ColumnInfo(string outputName, params string[] inputNames)
+                : this(outputName, GetPairs(inputNames))
+            {
+            }
+
+            private static IEnumerable<(string name, string alias)> GetPairs(string[] inputNames)
+            {
+                Contracts.CheckValue(inputNames, nameof(inputNames));
+                return inputNames.Select(name => (name, (string)null));
+            }
+
+            public ColumnInfo(string outputName, IEnumerable<(string name, string alias)> inputs)
+            {
+                Contracts.CheckNonEmpty(outputName, nameof(outputName));
+                Contracts.CheckValue(inputs, nameof(inputs));
+                Contracts.CheckParam(inputs.Any(), nameof(inputs), "Can not be empty");
+
+                foreach (var (name, alias) in inputs)
```

Comment:
>alias [](start = 36, length = 5)

Can we have a test in which we use these aliases, or maybe some kind of ///summary on top of this constructor with description of aliases.
Right now it's mystery for me, and any potential user.